gravitee-apim-console-webui/src/management/api/list/apis-list.component.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Component, Inject, OnInit } from '@angular/core';
+import { StateService } from '@uirouter/core';
+import { catchError, tap } from 'rxjs/operators';
+import { of } from 'rxjs';
+
+import { UIRouterState } from '../../../ajs-upgraded-providers';
+import { Api } from '../../../entities/api';
+import { PagedResult } from '../../../entities/pagedResult';
+import { GioTableWrapperFilters } from '../../../shared/components/gio-table-wrapper/gio-table-wrapper.component';
+import { ApiService } from '../../../services-ngx/api.service';
+
+type ApisTableDS = { id: string; name: string }[];
+
+@Component({
+  selector: 'apis-list',
+  template: require('./apis-list.component.html'),
+  styles: [require('./apis-list.component.scss')],
+})
+export class ApisListComponent implements OnInit {
+  displayedColumns = ['name'];
+
+  apisTableDSUnpaginatedLength = 0;
+  apisTableDS: ApisTableDS = [];
+  filters: GioTableWrapperFilters = {
+    pagination: { index: 1, size: 10 },
+    searchTerm: '',
+  };
+
+  constructor(@Inject(UIRouterState) private readonly $state: StateService, private readonly apiService: ApiService) {}
+
+  ngOnInit(): void {
+    this.apiService
+      .list()
+      .pipe(
+        tap((apisPage) => this.convert(apisPage)),
+        catchError(() => of(new PagedResult<Api>())),
+      )
+      .subscribe();
+  }
+
+  onEditActionClicked(_api: ApisTableDS[number]) {
+    this.$state.go('management.apis.detail.portal.general', { apiId: _api.id });
+  }
+
+  onAddApiClick() {
+    this.$state.go('management.apis.new');
+  }
+
+  private convert(apis: PagedResult<Api>) {
```

Comment:
WDYT to avoid this  function with ttl :  
```
        tap((apisPage) =>  {
        this.apisTableDS = toApisTableDS(apisPage)
        this.apisTableDSUnpaginatedLength = apisPage.page.total_elements
}),
        
```
?

Reply:
Done